app/models/allocation.rb

```diff
@@ -12,9 +16,25 @@ class Allocation < VersionedModel
     long: 'more_than_16',
   }
 
+  # TODO: implement statemachine, for now just allow an allocation to be cancelled
+  enum status: {
+    unfilled: ALLOCATION_STATUS_UNFILLED,
+    filled: ALLOCATION_STATUS_FILLED,
+    cancelled: ALLOCATION_STATUS_CANCELLED,
+  }
+
+  CANCELLATION_REASONS = [
+    MADE_IN_ERROR = 'made_in_error',
+    SUPPLIER_DECLINED_TO_MOVE = 'supplier_declined_to_move',
+    LACK_OF_SPACE = 'lack_of_space_at_receiving_establishment',
+    FAILED_TO_FILL_ALLOCATION = 'sending_establishment_failed_to_fill_allocation',
+    OTHER = 'other',
+  ].freeze
+
   belongs_to :from_location, class_name: 'Location'
   belongs_to :to_location, class_name: 'Location'
-  has_many :moves, inverse_of: :allocation, dependent: :destroy
+  has_many :moves, inverse_of: :allocation, dependent: :destroy, autosave: true
```

Comment:
need to set autosave to allow updated to moved to be saved, as only create is set by default